packages/core/src/lib/switch.svelte

```diff
@@ -1,132 +1,101 @@
+  aria-checked={on ? 'true' : 'false'}
+  {...$$restProps}
+  class={cx(
+    'flex items-center gap-2 outline-offset-2',
+    {
+      'cursor-not-allowed': disabled,
+    },
+    extraClasses
+  )}
+  on:click={toggle}
+  on:click|capture={handleDisabled}
+  on:keydown={handleDisabledKeydown}
 >
-  <div class="flex items-center gap-1.5">
-    {#if label}
-      <button
-        class="w-fit text-xs"
-        on:click={handleClick}>{label}</button
-      >
-    {/if}
-
-    {#if tooltip}
-      <Tooltip>
-        <div class="-mt-0.5">
-          <Icon name="information-outline" />
-        </div>
-        <div slot="description">
-          {tooltip}
-        </div>
-      </Tooltip>
-    {/if}
-  </div>
-
-  <button
-    on:click={handleClick}
-    type="button"
-    class={cx('flex items-center gap-2', {
-      'pointer-events-none cursor-not-allowed': disabled || readonly,
-    })}
-    role="switch"
-    aria-label={label}
-    aria-disabled={disabled ? true : undefined}
-    aria-checked={on ? 'true' : 'false'}
+  <div
+    class={cx(
+      'relative inline-flex h-5 w-11 flex-shrink-0 border duration-200 ease-in-out focus:outline-none motion-safe:transition-colors',
+      {
+        'border-gray-6 bg-gray-6': !on && !disabled,
+        'border-success-dark bg-success-dark': on && !disabled,
+        'border-gray-4 bg-gray-4 text-disabled-dark': disabled,
+      }
+    )}
   >
-    <div
+    <span
       class={cx(
-        'relative inline-flex h-5 w-11 flex-shrink-0 cursor-pointer border duration-200 ease-in-out focus:outline-none motion-safe:transition-colors',
+        'pointer-events-none relative ml-px mt-px inline-block h-4 w-4 transform bg-white ring-0 duration-200 ease-in-out motion-safe:transition-transform',
         {
-          'border-gray-4 bg-gray-4': disabled || readonly,
-          'border-gray-6 bg-gray-6': !on && !(disabled || readonly),
-          'border-success-dark bg-success-dark': on && !(disabled || readonly),
-          'text-disabled-dark': disabled,
+          'border-gray-4': disabled,
+          'translate-x-0': !on,
+          'translate-x-6': on,
         }
       )}
-    >
-      <span
-        class={cx(
-          'pointer-events-none relative ml-px mt-px inline-block h-4 w-4 transform bg-white ring-0 duration-200 ease-in-out motion-safe:transition-transform',
-          {
-            'border-gray-4': disabled || readonly,
-          }
-        )}
-        class:translate-x-0={!on}
-        class:translate-x-6={on}
-      />
-      <input
-        {disabled}
-        {readonly}
-        class="hidden"
-        type="checkbox"
-        checked={on}
-        bind:this={input}
-      />
-    </div>
+    />
+    <input
+      {disabled}
+      type="checkbox"
+      class="hidden"
+      bind:checked={on}
+    />
+  </div>
 
-    {#if variant === 'annotated'}
-      <p class="text-sm">{on ? 'on' : 'off'}</p>
-    {/if}
-  </button>
-</label>
+  {#if annotated}
```

Comment:
[Nit]: This `annotated` feels unnecessary if the caller explicitly passes `on` or `off`, can we throw a slot check in here?

```suggestion
  {#if annotated || $$slots.on || $$slots.off }
```